admin/listPurchases/listView.html

```diff
@@ -98,18 +98,29 @@
 
 const amountOfShownSubpages = 5;
 const entriesPerSubpage = 20;
-var amountOfSubpages;
+let amountOfSubpages;
 
 function generateSubPageNavigation(){
-    nav.innerHTML = '';
-    orderSearchBar.value ? amountOfSubpages = Math.ceil(entries.filter(currentOption.filter).filter(searchBarFilter).length / entriesPerSubpage) : amountOfSubpages = Math.ceil(entries.filter(currentOption.filter).length / entriesPerSubpage);
-    if(entryPageIndex >= amountOfSubpages) entryPageIndex = amountOfSubpages - 1;
+    if(orderSearchBar.value){
+        amountOfSubpages = Math.ceil(entries.filter(currentOption.filter).filter(searchBarFilter).length / entriesPerSubpage);
+    }else{
+        amountOfSubpages = Math.ceil(entries.filter(currentOption.filter).length / entriesPerSubpage);
```

Comment:
If you extract a function `getSubpageCount` here, you can simplify the code a lot by making it more linear:

```js
let entries = entries.filter(currentOption.filter);

if(orderSearchBar.value) {
  entries = entries.filter(searchBarFilter);
}

return Math.ceil(entries.length / entriesPerSubpage);
```